File: sorting-algorithms/python/bubblesort.py
import random


# Gerar array com valores aleatórios
def generate_array(size):
    new_array: list[int] = []
    for i in range(0, size):
        new_array.append(random.randint(0, size));
    return new_array


# implementação bubble sort algorithm
def bubble_sort(array):
    for i in range(0, len(array) - 1):
        for j in range(0, len(array) - i - 1):
            if array[j] > array[j + 1]:
                array[j], array[j + 1] = array[j + 1], array[j]


unsorted_array = generate_array(20)

print("Unsorted: ", unsorted_array)

bubble_sort(unsorted_array)

print("Sorted: ", unsorted_array)

File: sorting-algorithms/javascript/bubblesort.js

const MAX = 10;

/* organiza a array ao clicar no button */
function sort(inputTextArray) {
	
	var unsortedArray = translateToArray(inputTextArray.value);	
	
	/* organizar array */
	bubbleSort(unsortedArray);
	
	/* setar os valores da lista organizados no input */
	inputTextArray.value = unsortedArray.join(', ');
	
}

/* gera valore aleatórios */
function generate(inputTextArray) {
	let randomArray = [];	
	for (let i = 0; i < MAX; i++) {
		randomArray.push( Math.floor(Math.random() * MAX));
	}
	inputTextArray.value = randomArray.join(', ');
}

/* converte o texto do input para uma array de números */
function translateToArray(inputText) {
	
	var translatedArray = [];
	var textArray = inputText.replaceAll('\\s+', '').split(',');
	
	for (let i = 0; i < textArray.length; i++) {
		translatedArray.push(parseInt(textArray[i]));
	}
	
	return translatedArray;
}

/* implementação do bubble sort algorithm */
function bubbleSort(array) {
	for (let i = 0; i < array.length - 1; i++) {
		for (let j = 0; j < array.length - i - 1; j++) {
			if (array[j] > array[j + 1]) {
				let aux = array[j];
				array[j] = array[j + 1];
				array[j + 1] = aux;
			}
		}
	}
}

function getInputArray() {
	return document.getElementById("inputArray");
}
